### PR TITLE
build(854): switch frontend package manager from yarn to npm

### LINKED ISSUE #854: [SPIKE] Investigate alternative to yarn
**Description:**

Our JavaScript package manager `yarn@1.22.10` [is practically unmaintained](https://github.com/yarnpkg/yarn/issues/6953) since the release of `yarn@2.0.0`. This means that `yarn "v1"` will never release new updates or features beyond vulnerability fixes.

The maintainers recommend users migrate to [`yarn "v2"`](https://github.com/yarnpkg/berry), which, due to technical complexities in `v2`, may not be feasible for our project. `v2` was released with a change to the `node_modules` architecture that is fundamentally incompatible with our app.

This is illustrated in the below screenshot. Notice how `v1` still has 1.5 million weekly downloads and `v2` only has some 600. The likely cause is still the incompatibility with packages that will likely never be fixed.

Screen Shot 2021-04-15 at 3 36 33 PM

I formally recommend we migrate to [`npm@7.0.0`](https://github.blog/2021-02-02-npm-7-is-now-generally-available/.

* It supports deterministic builds, something `yarn v1` was created to solve.
* Supports workspaces, another thing `yarn` was created for.
* Is being actively developed by Microsoft themselves, and is unlikely to be abandoned like `yarn v1`.
* It has [improved performance](https://github.com/npm/benchmarks#example-usage) and beat `yarn` in some benchmarks. An oft-cited reason for using `yarn` over `npm` in the past was significant speed differences that are arguably negligible now.
* `npm` serves as *the official* JavaScript package manager, in that it is included with `Node.js` and doesn't require additional configuration or installations

**Acceptance Criteria:** 

- [ ] Decide on alternative
- [ ] Create issue to remove `yarn` and implement alternative

**Notes:**
- https://github.com/yarnpkg/yarn/issues/8583

### PR DESCRIPTION
**Pull request by jorgegonzalez**
### Summary of Changes
_Provide a brief summary of changes_

Pull request closes #854
_Acceptance criteria as stated in the issue_

### How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
```

1. Open http://localhost:3000/ and sign in.
1. Ensure frontend build proceeds as normal
1. Ensure there are no errors when running locally with Docker
1. Run `npm install` and `npm test` in the local frontend directory and ensure they succeed gracefully

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
	- [x]  Remove yarn and implement npm as alternative
	- [x] Have a new lockfile and package manager
	- [x] update documentation
	- [x] reviewed and tested by @ADPennington 

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [ ] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [x] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*
+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/Technical-Documentation/how-government-will-test-a11y.md)
_n/a_

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful deployment by assigning a 'Deploy with CircleCI - {env}' label*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [ ] If this PR introduces backend code, is that code documented both inline and overall?
+ [ ] If this PR introduces frontend code, is that code documented both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?
_no new issues detected_

### Deliverable 8: Context

> * Performance Standard(s): Code must be understandable and contextualized for the reviewers possess the knowledge and background necessary for analysis and constructive criticism to take place.

> * Acceptable Quality Level: Code submitted in the pull request has context.*
+ [x] Does this pull request contain sufficient inline comments providing relevant context for code snippets?
+ [x] Is the code understandable and lucid?
+ [x] Does this pull request provide background for why coding decisions were made?
+ [x] Can you as a reviewer explain and take ownership of these elements presented in this code review?

### CODE REVIEW

File: trufflehog-exclude-patterns.txt [hunk 1]
```diff
@@ -4,4 +4,5 @@
 (.*/)?static/index_files/analytics.js
 (.*/)?regexes.json
 (.*/)?yarn.lock
```
Comment (jorgegonzalez) [inline trufflehog-exclude-patterns.txt:6]:
```suggestion
```

**Review by abottoms-coder — APPROVED** [review]

**Comment by jorgegonzalez** [comment]
@ADPennington what is the QASP review status of this? This PR is blocking https://github.com/raft-tech/TANF-app/pull/1537 https://github.com/raft-tech/TANF-app/pull/1511 https://github.com/raft-tech/TANF-app/pull/1551 https://github.com/raft-tech/TANF-app/pull/1460

**Review by ADPennington — APPROVED** [review]
thanks @jorgegonzalez 🚀 

Some notes:
During the tabletop for this PR, we discussed the new ["security vulnerabilities"](https://app.circleci.com/pipelines/github/raft-tech/TANF-app/6638/workflows/9611ef80-5472-4ed4-9476-3bce02accb17/jobs/15423/parallel-runs/0/steps/0-106) that are being reported from ` npm audit` (by way of `npm ci`) as part of the frontend build.  

`npm audit` results can be unreliable ([ref](https://overreacted.io/npm-audit-broken-by-design/)), and it is unclear why running this report locally yields inconsistent [results](https://hackmd.io/SqhcWdmDSLSduBy5q-cSCw).  So, we decided that these results should be suppressed, since we rely on Dependabot for dependency management and, this tool is sourced by the same dB ([ref](https://github.blog/2021-10-07-github-advisory-database-now-powers-npm-audit/)).  So we should get the same information (if it is valid) via the Security tab of this repo.  This suppression is accounted for with the merge of this PR. cc: @abottoms-coder @lfrohlich

**Comment by jorgegonzalez** [comment]
Good point on the GitHub Advisories db being the same as the npm audit one 👍

**Pull request merged**